### PR TITLE
Added temperature argument to the random sampler class

### LINKED ISSUE #947: Add a temperature argument to the base sampler class
We would like our samplers to support a [temperature](https://medium.com/mlearning-ai/softmax-temperature-5492e4007f71) argument that allows tightening or loosening the probability distribution.

We can add this pretty simply in the base class by dividing our logits by a temperature parameter [here](https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/samplers/sampler.py#L111-L112) before applying the softmax.

Checklist: 
 - [ ] Sampler takes in a `temperature` argument in `__init__`, adds it to `self`, and populates it in `get_config()`.
 - [ ] In `__call__`, use `self.temperature` to scale the logits.
 - [ ] Beam sampler will need to do the same (as it overrides call).
 - [ ] Add unit tests. E.g. [this test](https://github.com/keras-team/keras-nlp/blob/794c4b18084a1fdf04cea0a2cae4889dbadd8516/keras_nlp/samplers/top_k_sampler_test.py#L82-L95 ) could be adapted to run with a very low temperature, and assert that all output ids are zero.

### PR DESCRIPTION
**Pull request by TheAthleticCoder**
Resolves #947. 
Noticed that we now have a Random Sampler Class. After merging #951, merging this PR will add the temperature argument to the Random Sampler as well.

### CODE REVIEW

**Comment by mattdangerw** [comment]
I think we can just fold this change into https://github.com/keras-team/keras-nlp/pull/951!
